iina/ThumbnailCache.swift

```diff
@@ -158,20 +158,25 @@ class ThumbnailCache {
     // data blocks
     while file.offsetInFile != eof {
       // length
-      let blockLength: Int64 = file.read(type: Int64.self)
+      guard let blockLength: Int64 = file.read(type: Int64.self) else {
+        Logger.log("Cannot read image length. Cache file will be deleted.", level: .warning, subsystem: subsystem)
+        file.closeFile()
+        delete(pathURL)
+        return nil
+      }
```

Comment:
I feel like this doesn't need two specific checks. How about

```suggestion
      guard let blockLength = file.read(type: Int64.self),
          let timestamp = file.read(type: Double.self) else {
        Logger.log("Cannot read image header. Cache file will be deleted.", level: .warning, subsystem: subsystem)
        file.closeFile()
        delete(pathURL)
        return nil
      }
```

Reply:
Fixed.